front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -125,3 +139,145 @@ export function disabledTextColumn(
+  previousRowData: PathWaypointRow,
+  op: { fromRowIndex: number },
+  allWaypointsLength: number
+): PathWaypointRow {
+  const newRowData = { ...rowData };
+  if (
+    newRowData.departure &&
+    newRowData.arrival &&
+    (!isEqual(newRowData.arrival, previousRowData.arrival) ||
+      !isEqual(newRowData.departure, previousRowData.departure))
+  ) {
+    newRowData.stopFor = String(
+      durationInSeconds(
+        time2sec(newRowData.arrival.time || ''),
+        time2sec(newRowData.departure.time || '')
+      )
+    );
+  }
+  if (!newRowData.stopFor && op.fromRowIndex !== allWaypointsLength - 1) {
+    newRowData.onStopSignal = false;
+  }
+  newRowData.isMarginValid = !(
+    newRowData.theoreticalMargin && !marginRegExValidation.test(newRowData.theoreticalMargin)
+  );
+  if (newRowData.isMarginValid && op.fromRowIndex === 0) {
+    newRowData.arrival = undefined;
+    // As we put 0% by default for origin's margin, if the user removes a margin without
+    // replacing it to 0% (undefined), we change it to 0%
+    if (!newRowData.theoreticalMargin) {
+      newRowData.theoreticalMargin = '0%';
+    }
+  }
+  return newRowData;
+}
+
+/**
+ * This function goes through the whole array of path waypoints
+ * and updates the number of days since departure.
+ */
+export function updateDaySinceDeparture<T extends Pick<PathWaypointRow, 'arrival' | 'stopFor'>>(
+  pathWaypointRows: T[],
+  startTime?: IsoDateTimeString,
+  keepFirstIndexArrival?: boolean
+): T[] {
+  let currentDaySinceDeparture = 0;
+  let previousTime = startTime ? datetime2sec(new Date(startTime)) : Number.NEGATIVE_INFINITY;
+
+  return pathWaypointRows.map((pathWaypoint, index) => {
+    const { arrival, stopFor } = pathWaypoint;
+
+    const arrivalInSeconds = arrival?.time ? time2sec(arrival.time) : null;
+    let formattedArrival: TimeExtraDays | undefined;
+    if (arrivalInSeconds) {
+      if (arrivalInSeconds < previousTime) {
+        currentDaySinceDeparture += 1;
+        formattedArrival = {
+          time: arrival!.time,
+          daySinceDeparture: currentDaySinceDeparture,
+          dayDisplayed: true,
+        };
+      } else {
+        formattedArrival = {
+          time: arrival!.time,
+          daySinceDeparture: currentDaySinceDeparture,
+        };
+      }
+      previousTime = arrivalInSeconds;
+    }
+
+    let formattedDeparture: TimeExtraDays | undefined;
+    if (stopFor && arrivalInSeconds) {
+      const departureInSeconds = (arrivalInSeconds + Number(stopFor)) % SECONDS_IN_A_DAY;
+      const isAfterMidnight = departureInSeconds < previousTime;
+      if (isAfterMidnight) {
+        currentDaySinceDeparture += 1;
+        formattedDeparture = {
+          time: secToHoursString(departureInSeconds, true),
+          daySinceDeparture: currentDaySinceDeparture,
+          dayDisplayed: true,
+        };
+      } else {
+        formattedDeparture = {
+          time: secToHoursString(departureInSeconds, true),
```

Comment:
Same as previous comment

```suggestion
          time: secToHoursString(departureInSeconds, WITH_SECONDS),
          daySinceDeparture: currentDaySinceDeparture,
          dayDisplayed: true,
        };
      } else {
        formattedDeparture = {
          time: secToHoursString(departureInSeconds, WITH_SECONDS),
```

Reply:
https://github.com/OpenRailAssociation/osrd/pull/8334#discussion_r1759003784